src/basic/fd-util.c

```diff
@@ -822,6 +822,32 @@ int fd_reopen(int fd, int flags) {
         return new_fd;
 }
 
+int fd_reopen_propagate_flags(int fd, int flags, int propagate_flags) {
+        /* Invokes fd_reopen(fd, flags),
+         * but propagates selected flags if set on the original fd.
+         *
+         * You should use this for example if the original fd potentially is O_APPEND,
+         * otherwise we get rather "unexpected" behavior.
+         * Unless you intentionally want to overwrite pre-existing data,
+         * and have your output overwritten by the next guy.
+         *
+         * Used for "systemd-run --pty >> some-log", but may be used
+         * to propagate other flags as well, if there is a use case. */
+
+        /* Whitelist supported flags here.
+         * For now, the only supported flag is O_APPEND. */
+        const int supported_flags_mask = O_APPEND;
```

Comment:
uh, we generally don't use const for local values in our codebase. the compiler can figure this out on its own, and it shows up in no interface to anything outside, hence there's nothing gained from doing this.. please drop the "const"

Reply:
I've been reading along, having been underwhelmed by the quality of the code in general, but I was not going to say anything at first. But now, this has become a little bit too weird not to say something.

It may not be too relevant in this particular case, but  _**generally never**_ using `const` for local variables sounds like a horribly nonsensical rule. The main reason why `const` exists is not for the compiler to do things faster, it is for the developers to avoid accidentally changing things that are not supposed to change once established.

Also, looking at the coding standard linked above, I'm wondering: did anyone bother to read any existing best practices and codings standards - and the rationale why some things should be done the recommended way - before writing this document, or did you just turn random habits, including lots of bad ones, into rules?

Reply:
@raltnoeder you are entitled to your opinions. But let me just say that if you think that https://github.com/raltnoeder/cdsaext/blob/master/src/bsearch.c is the epitome of C coding style in your opinion then maybe we just have different opinions on style?

Our coding guidelines are pretty close (but also a bit different) to Linux kernel style. i'd invite you to have a coding style discussion with kernel people first, before you come back to complain about ours.

But anyway, please do not derail discussions like this, especially if you have never contributed a thing to the project. Thank you for understanding.

Reply:
I was not just talking about style, but even more important things like variable naming, code structure & control flow, clarity of the code & maintainability, etc.

I would like to point out that this has very little to do with **_my_** opinion. The way I do things is by following the recommendations of lots of people who have worked on projects that have a defect rate that is several orders of magnitude better than the defect rate of the Linux kernel, and who all came to the same conclusions about how things should be done.